### ISSUE TITLE
Sometimes failed to install: " Could not resolve org.nrg:parent:1.7.5."

### ISSUE THREAD

**Issue by willfurnass** [NrgXnat/xnat-pipeline-engine#4]
```
Could not resolve org.nrg:parent:1.7.5.  Required by: project
Could not get resource 'http://maven.imagej.net/content/groups/public/org/nrg/parent/1.7.5/parent-1.7.5.pom'
```

However, this is intermittent.  Any suggestions?

**Comment by rherrick** [comment]
Yeah, this is really quite annoying. The problem is that, for some reason, whoever’s running maven.scijava.net (or at least someone with write permissions there) has created folders for a number of our libraries, e.g. you can go to https://maven.scijava.org/content/groups/public/org/nrg/parent/1.7.5. There’s nothing in there, but that folder exists. Unfortunately, the version of Gradle on which the 1.7.5 pipeline is built isn’t smart enough to figure out that it should move onto the next repository in that particular case. The good news is that newer versions of Gradle do the right thing and work just fine with the pipeline’s build.gradle configuration. If you want to try that fix, it’s pretty simple:


  1.  Install Gradle<https://gradle.org/install>. I use sdkman<https://sdkman.io/> to manage a lot of my dev tools, so I install through that, Homebrew is also good, apt and yum are suboptimal because the version of Gradle is pretty old on the standard repos, e.g. it’s at version 4.4.1 on the repos for Ubuntu 18.04. You should install the latest version, but it has to be at least 5.x (I haven’t tested this fix robustly with all of the 5.x releases but can verify that 5.4 and 5.6.2 work properly).
  2.  In the pipeline installer folder (where gradlew and build.gradle are located), run this command:
$ gradle wrapper

That will update the wrapper script, properties, and jar file.

  1.  Now run the install again:

$ ./gradlew

That should work around the problem. The next release of the pipeline engine will have the wrapper already upgraded.

--
Rick Herrick
Sr. Programmer/Analyst
Neuroinformatics Research Group
Washington University School of Medicine
Phone: +1 (314) 273-1645
